app/utils/DateErrorFormatter.scala

```diff
@@ -16,12 +16,26 @@
 
 package utils
 
+import play.api.data.FormError
 import play.api.i18n.Messages
 
 object DateErrorFormatter {
 
   def formatArgs(args: Seq[Any])(implicit messages: Messages): Seq[String] = {
-    args.map(arg => messages(s"date.$arg").toLowerCase)
+    val dateArgs = Seq("day", "month", "year")
+    args.map(arg => if (dateArgs.contains(arg)) messages(s"date.$arg").toLowerCase else arg.toString)
+  }
+
+  def addErrorClass(error: Option[FormError], dateArg: String): String = {
```

Comment:
Improvement of this bit suggestion:
  def addErrorClass(error: Option[FormError], dateArg: String): String =
    error.map { e =>
      if (e.args.isEmpty || e.args.contains(dateArg)) s"govuk-input--error"
      else ""
    }.getOrElse("")

Reply:
Thanks!